core/camel-core-model/src/main/java/org/apache/camel/model/dataformat/JsonDataFormat.java

```diff
@@ -601,6 +614,16 @@ public Builder contentTypeHeader(boolean contentTypeHeader) {
             return this;
         }
 
+        public Builder dateFormatPattern(String dateFormatPattern) {
```

Comment:
Can you add javadoc with: To configure the date format while marshall or unmarshall Date fields in JSON using Gson

to the builder methods

Reply:
And frankly we likely forgot to do that on other builder methods. If you spot those, you are welcome to send another PR to update those.

Reply:
Addressed